engine/src/test/java/org/hibernate/validator/test/constraintvalidator/ConstraintValidatorPayloadTest.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.test.constraintvalidator;
+
+import static java.lang.annotation.ElementType.ANNOTATION_TYPE;
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+import static org.hibernate.validator.testutil.ConstraintViolationAssert.assertNoViolations;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import javax.validation.Constraint;
+import javax.validation.ConstraintValidatorContext;
+import javax.validation.Payload;
+import javax.validation.Validator;
+import javax.validation.metadata.ConstraintDescriptor;
+
+import org.hibernate.validator.HibernateValidatorConfiguration;
+import org.hibernate.validator.HibernateValidatorFactory;
+import org.hibernate.validator.cfg.ConstraintMapping;
+import org.hibernate.validator.constraintvalidation.HibernateConstraintValidator;
+import org.hibernate.validator.constraintvalidation.HibernateConstraintValidatorInitializationContext;
+import org.hibernate.validator.testutil.TestForIssue;
+import org.hibernate.validator.testutils.ValidatorUtil;
+import org.testng.annotations.Test;
+
+/**
+ * @author Matthias Kurz
+ */
+@TestForIssue(jiraKey = "HV-1529")
+public class ConstraintValidatorPayloadTest {
+
+	@Test
+	public void testHibernateValidatorContextPayload() {
+		HibernateValidatorConfiguration configuration = getConfiguration();
+
+		final Validator validator = configuration.buildValidatorFactory().unwrap( HibernateValidatorFactory.class )
+				.usingContext().constraintValidatorPayload( Integer.class )
+				.getValidator();
+
+		assertNoViolations(	validator.validate( new PayloadDummyEntity( Integer.class ) ) );
```

Comment:
Minor formatting issue here.

Reply:
Ah, wondering this didn't trigger a checkstyle error...

Reply:
Yeah, it's unfortunate but IIRC Checkstyle is only picky about having at least one whitespace after the parenthesis, it doesn't trigger an error if there are more. Probably because you could have had a new line then several tabs if you wanted to indent it differently and it would still respect the rule.